### ISSUE TITLE
'_' is assigned a value but never used when destructuring array

### ISSUE THREAD

**Issue by odedniv** [eslint/eslint#10444]
**Tell us about your environment**

* **ESLint Version:** 4.19.1
* **Node Version:** 9.8.0
* **npm Version:** 6.0.1

**What parser (default, Babel-ESLint, etc.) are you using?**
None

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  extends: 'eslint:recommended',
  parserOptions: {
    ecmaVersion: 2017,
    sourceType: 'module'
  },
  env: {
    browser: true,
    commonjs: true,
    es6: true,
    node: true,
    mocha: true
  },
  plugins: [
    'mocha',
  ],
  rules: {
    'indent': ['error', 2, { SwitchCase: 1 }],
    'linebreak-style': ['error', 'unix'],
    'semi': ['error', 'always'],
    'no-console': 'off',
    'eqeqeq': ['error', 'always'],
    'mocha/no-exclusive-tests': 'error',
  }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
let [_, a] = [1, 2];
console.log(a);
```

```bash
./node_modules/.bin/eslint src
```

**What did you expect to happen?**
Not report an error.
When `_` is used in function arguments, it is ignored and not reported as unused variable. I also tried adding this rule and it still failed (yaml):

```yaml
rules:
  no-unused-vars: [error, argsIgnorePattern: ^_]
```

**What actually happened? Please include the actual, raw output from ESLint.**

```
89:10  error  '_' is assigned a value but never used  no-unused-vars
```

**Comment by not-an-aardvark** [comment]
Thanks for the report. Have you tried using `varsIgnorePattern` rather than `argsIgnorePattern`? I think `argsIgnorePattern` only applies to function parameters.

It's also worth noting that you don't need to create an unused variable in this code -- you can simply omit the variable:

```js
let [, a] = [1, 2];
console.log(a);
```

**Comment by odedniv** [comment]
Right, so `eslint` is right to warn me here because it's really unnecessary. Thanks a lot!
